flink-runtime/src/main/java/org/apache/flink/runtime/util/StateHandleStoreUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.util;
+
+import org.apache.flink.runtime.persistence.StateHandleStore;
+import org.apache.flink.runtime.state.StateObject;
+import org.apache.flink.util.ExceptionUtils;
+import org.apache.flink.util.InstantiationUtil;
+
+/**
+ * {@code StateHandleStoreUtils} collects utility methods that might be usefule for any {@link
+ * StateHandleStore} implementation.
+ */
+public class StateHandleStoreUtils {
+
+    /**
+     * Serializes the passed {@link StateObject} and discards the state in case of failure.
+     *
+     * @param stateObject the {@code StateObject} that shall be serialized.
+     * @return The serialized version of the passed {@code StateObject}.
+     * @throws Exception if an error occurred during the serialization. The corresponding {@code
+     *     StateObject} will be discarded in that case.
+     */
+    public static byte[] serializeStateHandle(StateObject stateObject) throws Exception {
```

Comment:
I think it would be more consistent to have a symmetric method in this class to deserialize handle.
`ZooKeeperStateHandleStore` currently calls `InstantiationUtil.deserializeObject` for that.

Reply:
I thought about the same thing but hesitated since it was only a one-liner. I went ahead with your proposal and even added the missing `StateObjectUtilsTest`.